app/src/main/java/com/lagradost/cloudstream3/ui/search/SearchFragment.kt

```diff
@@ -315,16 +315,29 @@ class SearchFragment : Fragment() {
                         arrayAdapter.notifyDataSetChanged()
                     }
 
-                    val selectedSearchTypes = DataStoreHelper.searchPreferenceTags
-
                     bindChips(
-                        binding.tvtypesChipsScroll.tvtypesChips,
+                        selectMainpageBinding.tvtypesChipsScroll.tvtypesChips,
                         selectedSearchTypes,
-                        TvType.values().toList()
+                        validAPIs.flatMap { api -> api.supportedTypes }.distinct()
                     ) { list ->
                         updateList(list)
+
+                        // refresh selected chips in main chips
+                        if (selectedSearchTypes.toSet() != list.toSet()) {
+                            selectedSearchTypes.clear()
+                            selectedSearchTypes.addAll(list)
+                            bindChips(
+                                binding?.tvtypesChipsScroll?.tvtypesChips,
+                                selectedSearchTypes,
+                                validAPIs.flatMap { api -> api.supportedTypes }.distinct()
+                            ) {
+                                // This already handled in another bindChips. Do nothing here!
```

Comment:
Nope, this will break the logic as the callback (`button?.setOnCheckedChangeListener`) will be set to Unit and the buttons will stop working. If you want to keep the prev callback make the callback nullable

Reply:
updateChips was already there. use it now. pls chk.